python/paddle/fluid/tests/unittests/test_conv2d_int8_mkldnn_op.py

```diff
@@ -0,0 +1,223 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+import unittest
+import numpy as np
+
+import paddle.fluid.core as core
+from op_test import OpTest
+from test_conv2d_op import conv2d_forward_naive, TestConv2dOp
+
+
+def conv2d_forward_refer(input, filter, group, conv_param):
+    out, in_n, out_h, out_w, out_c = conv2d_forward_naive(input, filter, group,
+                                                          conv_param)
+    out_tmp = np.zeros((in_n, out_h, out_w, out_c))
+    for n in range(in_n):
+        for i in range(out_h):
+            for j in range(out_w):
+                for m in range(out_c):
+                    out_tmp[n, i, j, m] = out[n, m, i, j]
+    return out_tmp.reshape(in_n, out_c, out_h, out_w)
+
+
+class TestConv2dInt8Op(TestConv2dOp):
+    def setUp(self):
+        self.op_type = "conv2d"
+        self.use_cudnn = False
+        self.exhaustive_search = False
+        self.use_cuda = False
+        self.use_mkldnn = False
+        self.data_format = "AnyLayout"
+        self.weighttype = np.float32
+        self.use_mkldnn = True
+        self.init_group()
+        self.init_dilation()
+        self.init_test_case()
+        self.init_dtype()
+
+        conv2d_param = {
+            'stride': self.stride,
+            'pad': self.pad,
+            'dilation': self.dilations
+        }
+
+        filter = np.random.random(self.filter_size).astype(self.weighttype)
+        if self.srctype == np.uint8:
+            input = np.random.randint(0, 10,
+                                      self.input_size).astype(self.srctype)
+        else:
+            input = np.random.randint(-5, 5,
+                                      self.input_size).astype(self.srctype)
+            input_shift = (np.ones(self.input_size) * 128).astype(np.uint8)
+
+        if self.srctype == np.int8:
+            output1 = conv2d_forward_refer(
+                np.round((input.astype(np.int32) + input_shift) *
+                         self.scale_in).astype(np.int32),
+                np.round(filter * self.scale_weights[0] * 0.5).astype(np.int32),
+                self.groups, conv2d_param).astype(np.float32) * (
+                    self.scale_out / (self.scale_in * self.scale_weights[0] *
+                                      0.5))
+            output2 = conv2d_forward_refer(
+                np.round((input_shift) * self.scale_in).astype(np.int32),
+                np.round(filter * self.scale_weights[0] * 0.5).astype(np.int32),
+                self.groups, conv2d_param).astype(np.float32) * (
+                    self.scale_out / (self.scale_in * self.scale_weights[0] *
+                                      0.5))
```

Comment:
line71-74 is the same as line 77-80, could you reuse it? 
```
filter, group, conv_param = np.round(filter * self.scale_weights[0] * 0.5).astype(np.int32),
                self.groups, conv2d_param).astype(np.float32) * (
                    self.scale_out / (self.scale_in * self.scale_weights[0] *
                                      0.5))
output1 = conv2d_forward_refer(xxx，filter, group, conv_param）
```

Reply:
I think these reuse is unreadable, because conv2d_forward_refer is a function, and these codes are the parameters of it, we don't need to extract parameters to another function. What do you think?